### PR TITLE
Update compaction techniques

### LINKED ISSUE #47: Update compaction techniques
After updating accumulo-proxy to use accumulo 2.1, the usage of compaction also has to be updated. Currently, compaction strategies are being used which is marked for removal and should be updated.

### PR DESCRIPTION
**Pull request by DomGarguilo**
This PR aims to replace the deprecated compaction methods with the new ones.

Fixes #47 

* a new thrift object was added for the PluginConfig 
* The compaction test was replaced with a test for supplying a custom `CompactionSelector`

I'm not sure that this is the best place to have the `CompactionSelector` as it is only used for testing so I will probably move it but just wanted to get something working together first.

I also plan on looking into adding a test for custom `CompactionConfigurer` too.

### CODE REVIEW

**Comment by ctubbsii** [comment]
I don't know enough about the new compaction configuration to review this. I requested a review from @keith-turner or @dlmarion , who I think could assess it better.

File: src/main/java/org/apache/accumulo/proxy/ProxyServer.java [hunk 1]
```diff
@@ -403,12 +404,24 @@ public void compactTable(ByteBuffer login, String tableName, ByteBuffer startRow
...
+      }
+
+      if (configurerConfig != null) {
+        Map<String,String> options = configurerConfig.options == null ? Map.of()
+            : configurerConfig.options;
+
+        org.apache.accumulo.core.client.admin.PluginConfig cpc = new org.apache.accumulo.core.client.admin.PluginConfig(
+            configurerConfig.getClassName(), options);
+
+        compactionConfig.setConfigurer(cpc);
```
Comment (dlmarion) [inline src/main/java/org/apache/accumulo/proxy/ProxyServer.java:424]: This new logic looks correct, appears to match the code in `CompactCommand.setupConfigurableCompaction`.

**Review by dlmarion — APPROVED** [review]

**Comment by DomGarguilo** [comment]
Everything looks good here to me and the tests pass when run with my IDE but when running in the terminal, there is an error `Security sealing violation: package org.apache.accumulo.proxy is sealed`. Not sure exactly whats going on here but I'm looking at moving the `CompactionSelector` that is currently in the test directory into its own Jar file.

**Review by ctubbsii — APPROVED** [review]
I fixed the merge conflicts and the jar sealing issue. The changes look okay to me, and it was previously approved, so I'll merge it. The ITs are still failing, but that's for another issue.

**Pull request merged**